test/integration/shells_int_test.go

```diff
@@ -0,0 +1,104 @@
+package integration
+
+import (
+	"runtime"
+	"testing"
+
+	"github.com/ActiveState/cli/internal/testhelpers/e2e"
+	"github.com/ActiveState/cli/internal/testhelpers/tagsuite"
+	"github.com/ActiveState/termtest"
+	"github.com/stretchr/testify/suite"
+)
+
+type ShellsIntegrationTestSuite struct {
+	tagsuite.Suite
+}
+
+func (suite *ShellsIntegrationTestSuite) TestShells() {
+	suite.OnlyRunForTags(tagsuite.Critical, tagsuite.Shell)
+
+	ts := e2e.New(suite.T(), false)
+	defer ts.Close()
+
+	var shells []string
+	if runtime.GOOS == "linux" {
+		shells = []string{"bash", "fish"}
+	} else if runtime.GOOS == "darwin" {
+		shells = []string{"bash", "zsh", "tcsh"}
+	} else if runtime.GOOS == "windows" {
+		shells = []string{"cmd"}
+	}
+
+	// Checkout the first instance. It doesn't matter which shell is used.
+	cp := ts.SpawnWithOpts(e2e.WithArgs("checkout", "ActiveState-CLI/small-python"))
+	cp.Expect("Checked out project")
+	cp.ExpectExitCode(0)
+
+	for _, shell := range shells {
+		var cp *termtest.ConsoleProcess
+		// Run the checkout in a particular shell.
+		args := "checkout ActiveState-CLI/small-python " + shell
+		if shell == "bash" {
+			cp = ts.SpawnInBash(args)
+		} else if shell == "zsh" {
+			cp = ts.SpawnInZsh(args)
+		} else if shell == "tcsh" {
+			cp = ts.SpawnInTcsh(args)
+		} else if shell == "fish" {
+			cp = ts.SpawnInFish(args)
+		} else if shell == "cmd" {
+			cp = ts.SpawnInCmd(args)
+		}
```

Comment:
This seems like it could benefit from some sort of helper function, you're basically repeating it on line 64.

Reply:
Thanks for the e2e.Shell tip. I've incorporated that idea into something more generic than I had. It's not as fancy as your suggestion, but it's an improvement at least.